Catappult/appcoins-wallet-android
# [feature/APPC-3056 ]Wallet backup state info

**What does this PR do?**

   Adds logic across the app to use the backup state of a wallet to show the backup notifications/triggers or not.
   Removes the Card notification from home, only the trigger stays on home (now with a 30 day cooldown on dismiss).
   Old users (with no backup info yet) will see the trigger unless the go from the recover flow, as we will put that information on the server when a user backups or recovers a wallet
  Fix backup warning on MyWallets, it now should disappear when a wallet has backup saved on the server side

**Database changed?**
   
   No

**Where should the reviewer start?**

- BackupTriggerPreferences.kt
- HomeViewModel.kt
- BackupSystemNotificationPreferences.kt

**How should this be manually tested?**

  - Recover a wallet and check the hasBackup field in the walletInfo endpoint as it should change after the recover if the field was false (example: https://apichain.dev.catappult.io/transaction/wallet/0x23337e9caaa3f6b18b1080c95c46135514901837/info) 
  - Check that the trigger didn't show when it is a wallet with no backup on the server.
  - Check in the MyWallets that a wallet with backup done wont show the backup warning

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/APPC-3056

**Code Review Checklist**
